Java/ds/StackUsingQueue.java

```diff
@@ -0,0 +1,96 @@
+/* Java Program to implement a stack using two queues
+by making the push operation costly*/
+
+import java.util.*;
+
+public class StackUsingQueue {
+    static class Stack {
+        // the 2 queues
+        Queue<Integer> qu1 = new LinkedList<>();
+        Queue<Integer> qu2 = new LinkedList<>();
+
+        // storing current size
+        int currentSize;
+
+        Stack() {
+            currentSize = 0;
+        }
+
+        void push(int num) {
+            currentSize++;
+
+            // first, push num into an empty qu2
+            qu2.add(num);
+
+            // now we push the other elements from qu1 to qu2
```

Comment:
Add similar comments to other functionality

Reply:
Done